### PR TITLE
slimeBoots and slimeSling

### PR DESCRIPTION
**Pull request by Quetz4l**
* removed debug logs and comments 
* Set bounce distance to 20 blocks (up or away)
* small fixes
* add 100 durability to slime sling and slime boots
* bs

PS. thanks OneEyeMaker for his help

### CODE REVIEW

File: src/main/java/tconstruct/gadgets/TinkerGadgets.java
Comment (BlueWeabo) [inline src/main/java/tconstruct/gadgets/TinkerGadgets.java:1]: Are you sure removing this file won't cause any errors in full pack? Have you tested that no TC addon uses it?

File: src/main/java/tconstruct/armor/items/slime/ItemSlimeSling.java [hunk 1]
```diff
@@ -93,12 +91,17 @@ public void onPlayerStoppedUsing(ItemStack stack, World world, EntityPlayer play
...
-            player.playSound(TinkerGadgets.resource("slimesling"), 1f, 1f);
+            player.playSound(resource("slimesling"), 1f, 1f);
             SlimeBounceHandler.addBounceHandler(player);
+            stack.damageItem(1, player);
             // TinkerCommons.potionSlimeBounce.apply(player);
         }
     }
 
+    private String resource(String res) {
+        return String.format("%s:%s", "tinker", res);
```
Comment (BlueWeabo) [inline src/main/java/tconstruct/armor/items/slime/ItemSlimeSling.java:102]: any reason this is here? as it was from TinkerGadgets.java, which would make sense that it should either be left there or be put in TConstruct.java

File: src/main/java/tconstruct/library/SlimeBounceHandler.java [hunk 2]
```diff
@@ -187,7 +187,7 @@ public static boolean hasSlimeBoots(EntityLivingBase entity) {
             for (int i = 1; i < 5; i++) {
                 ItemStack aBoots = entity.getEquipmentInSlot(i);
                 if (aBoots != null && aBoots.getItem() instanceof ItemSlimeBoots) {
-                    // TinkerGadgets.log.info("Found boots in slot "+i);
+                    // TConstruct.logger.info("Found boots in slot "+i);
```
Comment (BlueWeabo) [inline src/main/java/tconstruct/library/SlimeBounceHandler.java:190]: any reason you need to leave the debug code in here?

File: src/main/java/tconstruct/armor/TinkerArmor.java
Comment (BlueWeabo) [inline src/main/java/tconstruct/armor/TinkerArmor.java:1]: As you have edited some of this file, could you also remove any wildcard imports there are here?

File: src/main/java/tconstruct/TConstruct.java
Comment (BlueWeabo) [inline src/main/java/tconstruct/TConstruct.java:1]: Can you remove wildcard imports from the file if you can?

**Review by BlueWeabo — COMMENTED** [review]
Thanks for the work! Just some minor things to address

File: src/main/java/tconstruct/TConstruct.java
Comment (Quetz4l) [reply src/main/java/tconstruct/TConstruct.java:1]: ok

File: src/main/java/tconstruct/armor/items/slime/ItemSlimeSling.java [hunk 3]
```diff
@@ -58,7 +60,7 @@ public EnumAction getItemUseAction(ItemStack stack) {
 
     @Override
     public int getMaxItemUseDuration(ItemStack stack) {
-        return 72000;
+        return 50000;
```
Comment (OneEyeMaker) [inline src/main/java/tconstruct/armor/items/slime/ItemSlimeSling.java:63]: Why did you select this value?

**Review by OneEyeMaker — CHANGES_REQUESTED** [review]
I'm not completely happy with removal of Gadgets feature. Tinkers' Gadgets is special module for little useful things. And it should be there for such this like slime boots.
You also shouldn't mix removal/refactoring PRs with balancing ones: I think this PR should focus on balancing.

Also please recheck your math. With this PR slime sling becomes uncontrollable - in my play test it moves player about 50 blocks away after holding it little more than 1 second!

File: src/main/java/tconstruct/gadgets/TinkerGadgets.java
Comment (OneEyeMaker) [reply src/main/java/tconstruct/gadgets/TinkerGadgets.java:1]: Removal of this class already has side-effect: creative tab for slime boots has no icon.

File: src/main/java/tconstruct/armor/items/slime/ItemSlimeSling.java (hunk 3 above)
Comment (Quetz4l) [reply src/main/java/tconstruct/armor/items/slime/ItemSlimeSling.java:63]: To slightly speed up use and reduce the jump range, but yeah, it's still a long range

Comment (Quetz4l) [reply src/main/java/tconstruct/armor/items/slime/ItemSlimeSling.java:63]: I wanted to speed up a bit and reduce the strength of the jumps, but now I see that it doesn't reduce the strength of the jumps.

**Comment by Quetz4l** [comment]
what are the values for the jumps? Up and away?

File: src/main/java/tconstruct/armor/items/slime/ItemSlimeSling.java (hunk 3 above)
Comment (OneEyeMaker) [reply src/main/java/tconstruct/armor/items/slime/ItemSlimeSling.java:63]: This value represents maximum time (in game ticks) which item can be held (using right-click).
It should be equal 72000 ticks = 20 * 3600 seconds = 1 hour.

File: src/main/java/tconstruct/armor/items/slime/ItemSlimeSling.java [hunk 4]
```diff
@@ -71,13 +73,9 @@ public void onPlayerStoppedUsing(ItemStack stack, World world, EntityPlayer play
...
         int i = this.getMaxItemUseDuration(stack) - timeLeft;
-        float f = i / 20.0F;
-        f = (f * f + f * 2.0F) / 3.0F;
-        f *= 4f;
 
-        if (f > 6f) {
-            f = 6f;
-        }
+        float f = i / 5.0F;
+        f = Math.min((f * f + f * 2.0F) * (4.0F / 3.0F), 6.0F);
```
Comment (OneEyeMaker) [inline src/main/java/tconstruct/armor/items/slime/ItemSlimeSling.java:78]: I think these calculations cause 50+ blocks jumping. Increase divider of `f` from 5 to 20 (like in original code) or even more. Also you can play little bit with final multiplier of `f` (4).

File: src/main/java/tconstruct/armor/items/slime/ItemSlimeBoots.java [hunk 5]
```diff
@@ -116,7 +119,7 @@ public void onFall(LivingFallEvent event) {
             // velocity is lost on server players, but we dont have to defer the bounce
             double gravity = 0.2353455252;
             double time = Math.sqrt(living.fallDistance / gravity);
-            double velocity = gravity * time;
+            double velocity = gravity * time / 2;
```
Comment (OneEyeMaker) [inline src/main/java/tconstruct/armor/items/slime/ItemSlimeBoots.java:122]: I don't think that this value should be touched. Needs play-testing.

Comment (Quetz4l) [reply src/main/java/tconstruct/armor/items/slime/ItemSlimeBoots.java:122]: without it, the bounce increases every time and you go higher and higher.

File: src/main/java/tconstruct/armor/items/slime/ItemSlimeSling.java (hunk 4 above)
Comment (Quetz4l) [reply src/main/java/tconstruct/armor/items/slime/ItemSlimeSling.java:78]: I've rewired the formula, and I'm just wondering what length to put in

**Review by OneEyeMaker — APPROVED** [review]

**Pull request merged**